pkg/kubectl/stop.go

```diff
@@ -88,18 +91,25 @@ func (reaper *ReplicationControllerReaper) Stop(namespace, name string, timeout
 	if err != nil {
 		return "", err
 	}
+	ctrl, err := rc.Get(name)
+	if err != nil {
+		return "", err
+	}
 	if timeout == 0 {
-		rc, err := rc.Get(name)
-		if err != nil {
-			return "", err
-		}
-		timeout = Timeout + time.Duration(10*rc.Spec.Replicas)*time.Second
+		timeout = Timeout + time.Duration(10*ctrl.Spec.Replicas)*time.Second
 	}
-	retry := NewRetryParams(reaper.pollInterval, reaper.timeout)
-	waitForReplicas := NewRetryParams(reaper.pollInterval, timeout)
-	if err = scaler.Scale(namespace, name, 0, nil, retry, waitForReplicas); err != nil {
+	overlappingCtrl, err := getReplicationControllersForLabels(rc, labels.Set(ctrl.Spec.Selector))
+	if err != nil {
 		return "", err
 	}
+	sort.Sort(controller.OverlappingControllers(overlappingCtrl))
+	if overlappingCtrl[0].Name == name {
```

Comment:
I'm only going to handle the exact match scenario, as follows:
1. Get a list of overlapping controllers (controllers with labels that are a superset of this one)
2. Check if all of them are an exact match, if any aren't, throw the error 
3. If there are exact match controllers, perform a simple delete. If there are no exact match controllers, scale and delete.

An evolution of this algorithm is to check in step 2, if there is a controller with a subset of the labels. If there are we can avoid throwing the error because its pods are being controlled by the subset rc. I think it's better to keep it simple though.